pinot-common/src/main/java/org/apache/pinot/common/datatable/DataTableImplV4.java

```diff
@@ -19,35 +19,643 @@
 
 package org.apache.pinot.common.datatable;
 
+import com.google.common.primitives.Ints;
+import com.google.common.primitives.Longs;
+import java.io.ByteArrayOutputStream;
+import java.io.DataOutputStream;
 import java.io.IOException;
+import java.math.BigDecimal;
 import java.nio.ByteBuffer;
-import org.apache.pinot.common.datablock.RowDataBlock;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.pinot.common.datablock.DataBlockUtils;
+import org.apache.pinot.common.request.context.ThreadTimer;
+import org.apache.pinot.common.response.ProcessingException;
 import org.apache.pinot.common.utils.DataSchema;
+import org.apache.pinot.common.utils.RoaringBitmapUtils;
 import org.apache.pinot.spi.annotations.InterfaceStability;
+import org.apache.pinot.spi.utils.BigDecimalUtils;
+import org.apache.pinot.spi.utils.ByteArray;
+import org.roaringbitmap.RoaringBitmap;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
 
 
 /**
  * Datatable V4 Implementation is a wrapper around the Row-based data block.
  */
 @InterfaceStability.Evolving
-public class DataTableImplV4 extends RowDataBlock {
+public class DataTableImplV4 extends BaseDataTable {
```

Comment:
Is V4 same as V3 + null handling ? I am guessing this is still needed even after decoupling

Reply:
Yes this class is basically a copy